### PR TITLE
expanded vfx and stage stuff to modules

### PR DESCRIPTION
**Pull request by nolram88**
encapsulated transparency global variable and moved common methods to a new utility file.

### CODE REVIEW

**Comment by wbobeirne** [comment]
I don't really see the point of having both a `vfx.js` and a `vfxutil.js`, why not keep them all under `main/vfx` and just split out exports in to different files? For instance, what makes drawHexagon live in vfx instead of vfxutil? I'd rather it live in `main/vfx/drawHexagon.js` and import only what it needs to run.

**Comment by nolram88** [comment]
I didn't really consider going that far with this one. this was primarily to sort out the circular dependency. It can get pretty out of hand if I take it further for this pr.

**Comment by wbobeirne** [comment]
Fair enough. How do you feel about meeting half way and changing `main/vfxutil.js` to `main/vfx/util.js` to better communicate the hierarchy?

**Comment by nolram88** [comment]
yea i was going to split that file out anyway.

File: src/main/vfxutil.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  transparency = !transparency;
+}
+
+export function makeColour(r, g, b, a) {
+  // maybe some hsl too
+  if (transparency) {
+    return "rgba(" + r + "," + g + "," + b + "," + a + ")";
+  }
+  else {
+    return "rgb(" + r + "," + g + "," + b + ")";
```
Comment (wbobeirne) [inline src/main/vfxutil.js]: ES6 has string interpolation built in to the language:
```
return `rgba(${r},${g},${b},${a}`;
```
A little more succinct and readable.

File: src/main/vfxutil.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+export let transparency = true;
+
+export function getTransparency(){
+  return transparency;
+}
+export function toggleTransparency(){
+  transparency = !transparency;
+}
+
```
Comment (nolram88) [inline src/main/vfxutil.js]: should i treat this variable as its own entity and put these into a whole new file?

**Pull request merged**